sycl/test-e2e/Matrix/joint_matrix_bf16_fill_k_cache_impl.hpp

```diff
@@ -74,152 +74,258 @@ double joint_matmul(TOperand *A, TOperand *B, TResult *C, queue &q, int i) {
           auto m1 = it.get_local_id(0);
           auto n1 = it.get_local_id(1) / sgSize;
           auto sg = it.get_sub_group();
+#ifdef PREFETCH
+          size_t sgId = sg.get_group_id()[0];
+          // There are MCache2/MCache1 x NCache2/NCache1 subgroups: NumSGs
+          // PVC case: this is 8x4 subgroups
+          // BKM for PVC is to use prefetch of 8x32 for each subgroup
+          constexpr size_t prefRow = 8;
+          constexpr size_t prefCol = 32;
+      // All the SGs of one workgroup prefetch MCache2xKCache2 of A
+      // All the SGs of one workgroup prefetch KCache2xNCache2 of B
+      // PVC case: 256x32 of A and 32x256 of B
+      // For both A and B: each subgroup performs a prefetch of
+      // prefRow rows and prefCol cols at a time
+      // For A, the subgroups are distributed along the row dimension:
+      // PVC: A layed as MCache2/prefRow (256/32)
+      // For B: the subgroups are distributed along the column dimension
+      // PVC: NCache2/prefCol = 256/32 = 8 SGs on the column dimension and
+      // KCache2/prefRow = 32/8 = 4 SGs on the row dimension
+#ifdef VNNI
+          // In the VNNI case, each subgroup still gets prefRow x prefCol
+          // subgroups distribution become 16 on the row dimension and
+          // 2 on the  col dimension
+          // PVC case: NCache2*2/prefCol = 16, KCache2/prefRow   so NumSGs*2
```

Comment:
should it be like that? If yes, does it mean we need 2x SGs for each row, but 1/2x SGs for each column, so total number of SGs (NumSGs) is a constant?
```suggestion
          // PVC case: NCache2*2/prefCol = 16, (KCache2/2)/prefRow = 2, so NumSGs*2
```

Reply:
I meant double the subgroups are needed for the VNNI case. We needed 8 subgroups in the row major case for each row. In VNNI, we need 16. I will improve the wording